plugin/keyring_vault/vault_curl.cc

```diff
@@ -194,71 +197,71 @@ bool Vault_curl::setup_curl_session(CURL *curl)
+      (!vault_credentials_.get_vault_ca().empty() &&
+       (curl_res= curl_easy_setopt(
+            curl, CURLOPT_CAINFO,
+            vault_credentials_.get_vault_ca().c_str())) != CURLE_OK) ||
       (curl_res= curl_easy_setopt(curl, CURLOPT_USE_SSL, CURLUSESSL_ALL)) !=
           CURLE_OK ||
       (curl_res= curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L)) !=
           CURLE_OK ||
-      (curl_res= curl_easy_setopt(curl, CURLOPT_PROGRESSFUNCTION,
-                                  progress_callback)) != CURLE_OK ||
-      (curl_res= curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 0L)) !=
-          CURLE_OK ||
       (curl_res= curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT, timeout)) !=
           CURLE_OK ||
       (curl_res= curl_easy_setopt(curl, CURLOPT_TIMEOUT, timeout)) !=
           CURLE_OK ||
-      (curl_res = curl_easy_setopt(curl, CURLOPT_HTTP_VERSION,
-                                   (long)CURL_HTTP_VERSION_1_1)) !=
-          CURLE_OK)
+      (curl_res= curl_easy_setopt(curl, CURLOPT_HTTP_VERSION,
+                                  (long)CURL_HTTP_VERSION_1_1)) != CURLE_OK)
   {
-    logger->log(MY_ERROR_LEVEL, get_error_from_curl(curl_res).c_str());
+    logger_->log(MY_ERROR_LEVEL, get_error_from_curl(curl_res).c_str());
     return true;
   }
   return false;
 }
 
-bool Vault_curl::list_keys(Secure_string *response)
-{
+bool Vault_curl::do_list(const Secure_string &url_to_list, Secure_string *response) {
+
   CURLcode curl_res = CURLE_OK;
   long http_code = 0;
 
-  Thd_wait_end_guard thd_wait_end_guard;
-  (void)thd_wait_end_guard; // silence unused variable error
-
   CURL *curl = curl_easy_init();
-  if (curl == NULL)
-  {
-    logger->log(MY_ERROR_LEVEL, "Cannot initialize curl session");
+  if (curl == NULL) {
+    logger_->log(MY_ERROR_LEVEL, "Cannot initialize curl session");
     return true;
   }
   Curl_session_guard curl_session_guard(curl);
 
   if (setup_curl_session(curl) ||
-      (curl_res = curl_easy_setopt(curl, CURLOPT_URL, (vault_url + "?list=true").c_str())) != CURLE_OK ||
+      (curl_res = curl_easy_setopt(curl, CURLOPT_URL, url_to_list.c_str())) != CURLE_OK ||
       (curl_res = curl_easy_perform(curl)) != CURLE_OK ||
-      (curl_res = curl_easy_getinfo (curl, CURLINFO_RESPONSE_CODE, &http_code)) != CURLE_OK)
-  {
-    logger->log(MY_ERROR_LEVEL,
-                get_error_from_curl(curl_res).c_str());
+      (curl_res = curl_easy_getinfo (curl, CURLINFO_RESPONSE_CODE, &http_code)) != CURLE_OK) {
+    logger_->log(MY_ERROR_LEVEL, get_error_from_curl(curl_res).c_str());
     return true;
   }
-  if (http_code == 404)
-  {
-    *response = ""; // no keys found
+  if (http_code == 404) {
+    *response = ""; // list returned empty list
     return false; 
   }
   *response = read_data_ss.str();
   return http_code / 100 != 2; // 2** are success return codes
 }
 
+bool Vault_curl::list_mount_points(Secure_string *response) {
+  return do_list(vault_credentials_.get_vault_url() + "/v1/sys/mounts",
```

Comment:
Is it Ok to hardcode v1 in URL here?

Reply:
Yes, `/v1/` is totally fine here - see previous comment. Moreover, mount point secret engine version auto-detection mechanism has been completely reworked in one of the later commits and `Vault_curl::list_mount_points()` method has been removed.